### PR TITLE
Added an s-dependent IOTA lens example

### PR DESCRIPTION
**Pull request by cemitch99**
Added a segmented IOTA lens example (Python) using 18 segments with the full s-dependence of the focusing.

The number of segments can be adjusted as a parameter.

The diagnostic output for the nonlinear invariants (H,I) needs to be updated to the corresponding values of alpha, beta, etc.  This requires that Python bindings be added for these parameters.

### Tasks

- [x] Update Testing Routines (in CMake)
- [x] Update IOTA Example Documentation (README.rst) - can also be a separate one instead of extending the existing one
- [x] Expose Python bindings for nonlinear invariants diagnostics

### CODE REVIEW

File: examples/iota_lattice/run_iotalattice_sdep.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+# build lattice: first half, qe3, then mirror
+# modified to place nll as the first element
+# fmt:on
+sim.lattice.append(monitor)
+sim.lattice.extend(nll)
+sim.lattice.extend(lattice_after_nll)
+sim.lattice.append(qe3)
+lattice_after_nll.reverse()
+sim.lattice.extend(lattice_after_nll)
+sim.lattice.extend(dnll)
```
Comment (RTSandberg) [inline examples/iota_lattice/run_iotalattice_sdep.py]:
```suggestion
sim.lattice.append(dnll)
```

File: src/particles/diagnostics/NonlinearLensInvariants.H [hunk 2]
```diff
@@ -82,8 +82,8 @@ namespace impactx::diagnostics
             // convert transverse phase space coordinates to normalized units
             amrex::ParticleReal const xn = x/(m_cn*sqrt(m_beta));
             amrex::ParticleReal const yn = y/(m_cn*sqrt(m_beta));
-            amrex::ParticleReal const pxn = px*sqrt(m_beta)/m_cn + m_alpha*x;
-            amrex::ParticleReal const pyn = py*sqrt(m_beta)/m_cn + m_alpha*y;
+            amrex::ParticleReal const pxn = px*sqrt(m_beta)/m_cn + m_alpha*xn;
+            amrex::ParticleReal const pyn = py*sqrt(m_beta)/m_cn + m_alpha*yn;
```
Comment (cemitch99) [inline src/particles/diagnostics/NonlinearLensInvariants.H:87]: Fixed a typo here.  This only affects the diagnostics when alpha is nonzero.

File: examples/iota_lens/README.rst [hunk 3]
```diff
@@ -48,3 +48,61 @@ We run the following script to analyze correctness:
    .. literalinclude:: analysis_iotalens.py
       :language: python3
       :caption: You can copy this file from ``examples/iota_lens/analysis_iotalens.py``.
+
+
+.. _examples-iotalens-sdep:
+
+A nonlinear focusing channel based on the physical IOTA nonlinear magnet
+=========================================================================
```
Comment (ax3l) [inline examples/iota_lens/README.rst]:
```suggestion
A nonlinear focusing channel based on the physical IOTA nonlinear magnet
========================================================================
```

File: examples/iota_lattice/README.rst [hunk 4]
```diff
@@ -48,3 +48,55 @@ We run the following script to analyze correctness:
    .. literalinclude:: analysis_iotalattice.py
       :language: python3
       :caption: You can copy this file from ``examples/iota_lattice/analysis_iotalattice.py``.
+
+
+.. _examples-iotalattice-sdep:
+
+The full nonlinear lattice of the Fermilab IOTA storage ring
+=============================================================
```
Comment (ax3l) [inline examples/iota_lattice/README.rst]:
```suggestion
The full nonlinear lattice of the Fermilab IOTA storage ring
============================================================
```

**Review by ax3l — APPROVED** [review]
Thank you, looks great ! :+1:

**Pull request merged**